
syntax = "proto3";
package terragen;


enum MessageType {
  UNDEFINED = 0;
  LOGIN = 1;
  LOGIN_RESPONSE = 2;
  DISCONNECT = 3;
  CHAT = 4;
  CHAT_BROADCAST = 5;
  POSITION_UPDATE = 6;
  POSITION_UPDATE_BROADCAST = 7;
  BLOCK_UPDATE = 8;
  BLOCK_UPDATE_BROADCAST = 9;
  CHUNK_TRANSMIT = 10;
}


message EnvelopeModel {
  string Id = 1;
  int32 PocketId = 2;
  int32 PocketsCount = 3;
  MessageType Type = 4;
  bytes PayLoad = 5;
}